### PR TITLE
Add more tests for converting to Array and some other things

### PR DESCRIPTION
**Pull request by kshyatt**
cc @andreasnoack @Sacha0 @tkelman.

### CODE REVIEW

File: test/linalg/cholesky.jl [hunk 1]
```diff
@@ -62,24 +62,32 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
     @test all(x -> x ≈ √apos, cholfact(apos).factors)
     @test_throws ArgumentError chol(-one(eltya))
 
-    if eltya <: Real
+    if eltya <: BlasReal
```
Comment (andreasnoack) [inline test/linalg/cholesky.jl]: Not necessary

File: test/linalg/cholesky.jl [hunk 2]
```diff
@@ -62,24 +62,32 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
...
         @test inv(capds)*apds ≈ eye(n)
         @test abs((det(capds) - det(apd))/det(capds)) <= ε*κ*n
-    else
+        capds2 = cholfact!(copy(apds))
+        @test Array(capds2) ≈ Array(capds)
+    elseif eltya <: BlasComplex
         capdh = cholfact(apdh)
         @test inv(capdh)*apdh ≈ eye(n)
         @test abs((det(capdh) - det(apd))/det(capdh)) <= ε*κ*n
+        @test Array(cholfact!(copy(apdh))) ≈ Array(capdh)
```
Comment (andreasnoack) [inline test/linalg/cholesky.jl]: Maybe better just to test against `apdh`.

**Comment by kshyatt** [comment]
It is necessary because otherwise the Int symmetric matrix gets caught in
cholfact! and errors

**Comment by tkelman** [comment]
should still be testing BigFloat though on the existing ones

File: test/linalg/qr.jl [hunk 3]
```diff
@@ -38,107 +37,108 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
...
+                    q, r  = qra[:Q], qra[:R]
+                    @test_throws KeyError qra[:Z]
+                    @test q'*full(q, thin=false) ≈ eye(n)
+                    @test q*full(q, thin=false)' ≈ eye(n)
+                    @test q'*eye(n)' ≈ full(q, thin=false)'
+                    @test full(q, thin=false)'q ≈ eye(n)
+                    @test eye(n)'q' ≈ full(q, thin=false)'
+                    @test q*r ≈ a
+                    @test_approx_eq_eps a*(qra\b) b 3000ε
+                    @test Array(qra) ≈ a
```
Comment (Sacha0) [inline test/linalg/qr.jl]: Re. `Array(qra)`, please see the comment above regarding `full` translating to `AbstractArray` for factorizations.

File: test/linalg/qr.jl [hunk 4]
```diff
@@ -38,107 +37,108 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
...
+                    @test_throws KeyError qrpa[:Z]
+                    p = qrpa[:p]
+                    @test q'*full(q, thin=false) ≈ eye(n1)
+                    @test q*full(q, thin=false)' ≈ eye(n1)
+                    @test (UpperTriangular(eye(eltya,size(q,2)))*q')*full(q, thin=false) ≈ eye(n1)
+                    @test q*r ≈ (isa(qrpa,QRPivoted) ? a[1:n1,p] : a[1:n1,:])
+                    @test q*r[:,invperm(p)] ≈ a[1:n1,:]
+                    @test q*r*qrpa[:P].' ≈ a[1:n1,:]
+                    @test_approx_eq_eps a[1:n1,:]*(qrpa\b[1:n1]) b[1:n1] 5000ε
+                    @test Array(qrpa) ≈ a[1:5,:]
```
Comment (Sacha0) [inline test/linalg/qr.jl]: Re. `Array(qrpa)`, please see the comment above regarding `full` translating to `AbstractArray` for factorizations.

File: test/linalg/qr.jl [hunk 5]
```diff
@@ -38,107 +37,108 @@ for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
...
+                @testset "(Automatic) Thin (pivoted) QR decomposition" begin
+                    qrpa  = factorize(a[:,1:n1])
+                    q,r = qrpa[:Q], qrpa[:R]
+                    @test_throws KeyError qrpa[:Z]
+                    p = qrpa[:p]
+                    @test q'*full(q, thin=false) ≈ eye(n)
+                    @test q*full(q, thin=false)' ≈ eye(n)
+                    @test q*r ≈ a[:,p]
+                    @test q*r[:,invperm(p)] ≈ a[:,1:n1]
+                    @test Array(qrpa) ≈ a[:,1:5]
```
Comment (Sacha0) [inline test/linalg/qr.jl]: Re. `Array(qrpa)`, please see the comment above regarding `full` translating to `AbstractArray` for factorizations.

File: test/linalg/tridiag.jl [hunk 6]
```diff
@@ -120,6 +120,7 @@ for elty in (Float32, Float64, Complex64, Complex128, Int)
             x = Ts\vv
             @test x ≈ invFsv
             @test Array(AbstractArray(Tldlt)) ≈ Fs
+            @test Array(Array(Tldlt)) ≈ Fs
```
Comment (Sacha0) [inline test/linalg/tridiag.jl]: Re. the `Array(Array(...))`, simply `Array()`?

**Review by Sacha0 — COMMENTED** [review]
A few comments primarily regarding replacement of `full` follow below :). (I plan to write a note about the status of `full` replacement/deprecation elsewhere shortly.) Best!

**Comment by kshyatt** [comment]
Looks like an unrelated AV failure.

**Comment by Sacha0** [comment]
(Recapping an offline conversation: I do not feel strongly about the points I identified above. Please feel welcome to ignore them :).)

**Comment by tkelman** [comment]
`schurfact`, `svdfact`, `hessfact`, and `lufact` outputs should all be using `AbstractArray` rather than `Array`

**Comment by andreasnoack** [comment]
This is from before the migration to standard libraries to it can't really be rebased so it would be easier to start over.

**Pull request closed without merging**